spec/hyrax/transactions/apply_change_set_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+require 'spec_helper'
+require 'hyrax/transactions'
+require 'dry/container/stub'
+
+RSpec.describe Hyrax::Transactions::ApplyChangeSet do
+  subject(:tx)     { described_class.new }
+  let(:change_set) { Hyrax::ChangeSet.for(resource) }
+  let(:resource)   { build(:hyrax_work) }
+  let(:xmas)       { DateTime.parse('2018-12-25 11:30').iso8601 }
+
+  before { allow(Hyrax::TimeService).to receive(:time_in_utc).and_return(xmas) }
+
+  describe '#call' do
+    it 'is a success' do
+      expect(tx.call(change_set)).to be_success
+    end
+
+    it 'wraps a saved work' do
+      result = tx.call(change_set)
+
+      expect(result.value!).to be_persisted
```

Comment:
I'm not used to seeing a comparison of a "bang!" method return value. I'm assuming that's a Dry::Monad idiom.

Reply:
it's usual for `Dry::Monad`'s `Result` monads to access with handling for both the `Success` case (do something with the value) *and* the `Failure` case (log an error, recover, and/or raise). this is what's going on in `#value_or` calls; a bit like `Hash#fetch`'s "give me the value, or run this block" approach.

in the test suite, it's handy to just use this `#value!` method ("give met the value, or panic") since we expect a `Success` and are glad to see the test fail eagerly in any other case.

Reply:
Excellent. Thanks for the insight.